### DISCUSSION TITLE
This doesn't appear to be doing anything, how can I troubleshoot it?

### DISCUSSION THREAD

**Discussion by red8888** [tox-dev/sphinx-autodoc-typehints#281] [Q&A] [answered]
It doesn't appear to be doing anything. Is there a way I can debug it to see what its trying to do?

```
# conf.py
extensions = [
    "sphinx.ext.autodoc",
    "invocations.autodoc",
    "sphinx.ext.napoleon",
    # made sure to add _after_ napoleon
    "sphinx_autodoc_typehints",
]


# in my module
def derps(
    param: str = "sdlkfjskdf",
):
    """testing.

    Parameters
    ----------
    param : default "sdlkfjskdf"
        This is a pyinvoke task that can be called directly
    """
    print("lksdjflksjdfklsdjf")


# in an rst file
.. automodule:: my.module.name
    :members:
```

This is what I see:
<img width="948" alt="image" src="https://user-images.githubusercontent.com/6265399/210903672-79e671a8-086d-4d7d-9a68-0b983ffdc2ba.png">

I get no errors from the module on build.

Any ideas where to start with troubleshooting?

**Comment by gaborbernat** [comment, marked as answer]
Does that look like napoleon style? I don't think we support that; your docstring should look something like https://github.com/tox-dev/sphinx-autodoc-typehints/blob/main/tests/roots/test-dummy/dummy_module.py#L18-L23

**Comment by red8888** [comment]
Its numpy style which napoleon supports

Per this issue it should work if I have them ordered correctly: https://github.com/tox-dev/sphinx-autodoc-typehints/issues/15

**Comment by red8888** [comment]
AH HA! Default param is whats breaking this.

```
def derps(
    param_one: str = "abc",
    param_two: int = 123,
):
    """Some functions.

    Parameters
    ----------
    param_one :
        This is a description
    param_two : default=123
        This is a description
    """
    print("sldjflksdjflksdjf")
```

param_one includes the type and looks like this: `param_one (str) – This is a description`
but param_two looks like this: `param_two (default 123) – This is a description`.

So I guess two options would work for me:
* If I have specified the default in the docstring don't replace type with it include the type and the default
* Ideally I could leave out default in the docstring and have it generate not just the type but the default as well from the definition

**Comment by red8888** [comment]
Ok think I have this sorted. 

I'm able to use `sphinx_autodoc_defaultargs` in conjuction with this and get everything I want it seems.

Using both extensions and leaving out type and default info in the docstring will autogen both.

```
extensions = [
    "sphinx.ext.autodoc",
    "sphinx.ext.autosummary",
    "sphinx.ext.napoleon",
    "sphinx_autodoc_typehints",
    # include defaultargs ext too
    "sphinx_autodoc_defaultargs", 
]
```

Leave out type AND default values in docstrings:
```
def derps(
    param_one: str = "abc",
    param_two: int = 123,
):
    """Some functions.

    Parameters
    ----------
    param_one :
        This is a description
    param_two : 
        This is a description
    """
```

Looks like this when rendered:
<img width="646" alt="image" src="https://user-images.githubusercontent.com/6265399/210945607-70097f92-6928-492a-8fd0-e7f353183d4b.png">
